### ISSUE TITLE
Temporary fix for crashing

### ISSUE THREAD

**Issue by spikespaz** [d-language-server/dls#12]
When a part of DLS fails, such as parsing the file, this is likely not an error that should be considered fatal. Most of the time the file is just incomplete and the parser crashed reading it, and that shouldn't affect your workflow. Just silently ignore the error, and wait for the next file save to parse again. For now, keep the old information from the previous parse.

What about just ignoring parsing errors? As far as I am concerned, the only things that should break DLS completely is formatting. Not even symbol information (IntelliSense).

I don't need that stupid output log popping up when I forget a brace on a block, I shouldn't have to reload VS Code just to get it working again. Don't bother the user with fatal error messages that aren't DLS's fault.

**Comment by spikespaz** [comment]
@LaurentTreguier Well could we have an opt-in to steamroll dfmt and libdparse's errors? And I didn't say formatting should be allowed to break DLS, I don't think it should be allowed to break DLS. That's what I would like to see fixed, or at least suppressed.

**Comment by LaurentTreguier** [comment]
Catching errors is still really ugly though. But since it could be helpful (at least until I can upgrade dfmt and libdparse), I'll see if I can do it.

> As far as I am concerned, the only things that should break DLS completely is formatting

That does imply formatting could break DLS :/

**Comment by spikespaz** [comment]
@LaurentTreguier Yes it's ugly, but it should be a temporary measure. I would rather use some bad practices than have the server crash every time I make a typo.

And yeah I see what you're getting at, not sure what I was thinking when I worded that sentence.

**Comment by LaurentTreguier** [comment]
Version 0.14.3 will be available in a bit less than an hour. Both the VSCode extension and the Atom package now have a `d.init.catchErrors` setting that will catch any and all errors blindly.

**Comment by spikespaz** [comment]
@LaurentTreguier Thanks for implementing this! Hopefully it won't be needed much longer.
